### DISCUSSION TITLE
Any reason why still using Jetty 9.x and not 10 ?

### DISCUSSION THREAD

**Discussion by vhemery** [eclipse-emfcloud/emfcloud#99] [ModelServer] [answered]
Hello, with the recent target platform update, we moved from Eclipse 2021-03 to 2021-06.
Since Eclipse 2021-06 is using Jetty 10 (required by org.eclipse.equinox.http.jetty plugin), this leads to 2 versions of Jetty in the target platform : 9.4.43v20210629 and 10.0.5.

This seems to cause troubles with PDE in Eclipse.
I don't get why, Tycho can build the product, but Eclipse PDE has troubles resolving javax.servlet and/or jetty packages versioned dependencies. Maybe a PDE bug AFAICT.

But is there any reason why we can't use Jetty 10 in the model server ? Do we have special requirements on the 9.x version ? It seems to me much simpler to align the Jetty version with the rest of the environment.

**Comment by vhemery** [comment, marked as answer]
By looking at https://github.com/eclipse-emfcloud/emfcloud-modelserver/pull/124#issue-729682386 I guess Jetty 10 may break socket communication. If so, any clue why ?

Edit: the correct answer is the comment bellow by eneufeld, but I can't mark the reply as the answer. So keep reading this thread for the full answer.

**Reply by eneufeld** [reply]
Hi @vhemery , 
Jetty 10 and 11 do not break socket communication. The problem is rather, that Jetty does not release a p2 repository starting with version 10. So the consumption is tedious.
About the issue you see, the Eclipse Platform includes Jetty 10 basic bundles but without the whole websocket part (or we missed it) so we added Jetty 9 from their p2 update site. During the build tycho manages to resolve everything (I think we have version constraints somewhere) but PDE fails due two bundles being available.

**Reply by vhemery** [reply]
Thank you for taking the time to reply.
It looks to me like we can get basic Jetty 10 bundles from Eclipse P2 and other bundles we need from the Central Maven Repository, re-exposing them with the org.eclipse.emfcloud.modelserver.lib ...
On the Central Maven Repo, everything looks available. Some bundles have changed name, so this adds to the confusion.
E.g. https://mvnrepository.com/artifact/org.eclipse.jetty.websocket/websocket-api became https://mvnrepository.com/artifact/org.eclipse.jetty.websocket/websocket-jetty-api ; https://mvnrepository.com/artifact/org.eclipse.jetty.websocket/websocket-server became https://mvnrepository.com/artifact/org.eclipse.jetty.websocket/websocket-jetty-server

I too managed to build with Tycho. PDE prints errors, but the product can be launched and works nethertheless. PDE has always had issues with multiple bundle versions, and Tycho has always overcomed these, so no surprise here.
But if there is no known incompatibility, it's probably worth **someone** gives a shot at trying and migrating to Jetty 10 so we have a cleaner state.

**Someone** is always hard to find, since I guess we are all on tight schedules. But it's good to know that the improvement should be feasible. I'll let you know if I give it a try, but I can make no promise. If someone else wish to step up, do not hesitate to ask me for help if needed.
